indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -3132,7 +3153,7 @@ private void checkTaskDuration() throws ExecutionException, InterruptedException
           } else {
             DateTime earliestTaskStart = computeEarliestTaskStartTime(group);
 
-            if (earliestTaskStart.plus(ioConfig.getTaskDuration()).isBeforeNow()) {
+            if (earliestTaskStart.plus(ioConfig.getTaskDuration()).isBeforeNow() || group.getShutdownEarly()) {
```

Comment:
This looks like it is still subject to the stopTaskCount limit ... it really should not be.

Reply:
can't you just set stopTaskCount=0 to not have that config apply?

Reply:
Changing stopTaskCount requires a Supervisor change, which defeats the whole purpose ;) 

My clusters generally run with stopTaskCount=1 which works very well ... but for high taskCount jobs it is much more likely that one task will be cycling at any given time, which would render this command useless ... unless this command queues up the task for cycling, but I didn't think that logic is in the code.

Reply:
in this implementation it would respect the stopTaskCount and not stop the task until a "stop slot" is available.

Reply:
thinking about this some more i think it should probably just ignore stopTaskCount since the operator has manually requested the stop

Reply:
Please add a comment in the code explaining why we chose to ignore stopTaskCount when `shutdownEarly` is set to true. I think it makes sense to not respect it in this case, since stopTaskCount was meant to reduce task spikes during normal operation, and shutdownEarly is in response to an API call which is not a "normal" operation.